packages/wallet/src/cip30.ts

```diff
@@ -16,13 +19,49 @@ import { Logger, dummyLogger } from 'ts-log';
 import { SingleAddressWallet } from '.';
 import { cip30signData } from './KeyManagement/cip8';
 import { firstValueFrom } from 'rxjs';
+import { handleMessages } from '@cardano-sdk/cip30/dist/WebExtension';
 
 export type Cip30WalletDependencies = {
   logger?: Logger;
 };
 
+export enum Cip30ConfirmationCallbackType {
+  SignData = 'sign_data',
+  SignTx = 'sign_tx',
+  SubmitTx = 'submit_tx'
+}
+
+type SignDataCallbackParams = {
+  type: Cip30ConfirmationCallbackType.SignData;
+  data: {
+    addr: Cardano.Address;
+    payload: Cardano.util.HexBlob;
+  };
+};
+
+type SignTxCallbackParams = {
+  type: Cip30ConfirmationCallbackType.SignTx;
+  data: Cardano.NewTxBodyAlonzo;
+};
+
+type SubmitTxCallbackParams = {
+  type: Cip30ConfirmationCallbackType.SubmitTx;
+  // Sending the entire tx allows for future extensibility of FE without API modification
+  data: Cardano.NewTxAlonzo;
+};
+
+type CallbackConfirmation = (
```

Comment:
Params types would be good to export too
```suggestion
export type CallbackConfirmation = (
```